src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -914,6 +915,9 @@ public void handleStorageException(
           // A 401 indicates that the access token has expired,
           // we need to refresh the GCS client with the new token
           SnowflakeFileTransferAgent.renewExpiredToken(session, command, this);
+        } else if (se.getCode() == 401 && command != null) {
```

Comment:
Nit: checks could be less redundant written as:

```
if (se.getCode() == 401 && command != null) {
  if (session != null) {
    renew();
  } else {
    throw Exception();
  }
}
```
What you have is totally fine as well, the minimalism pedant side of me just feels better having at least written that out. :-)

Reply:
I like this better as well but just a nitpick